packages/compiler/src/server/serverlib.ts

```diff
@@ -339,6 +343,113 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function updateImportsOnFileChange(changes: FileEvent[]): Promise<void> {
```

Comment:
can pieces of this code split into separate functions like for example the filePathChangeMap resolution feels independent.